homeTask7/src/main/java/org/example/filter/FilterApplication.java

```diff
@@ -0,0 +1,34 @@
+package org.example.filter;
+
+import com.mysql.cj.util.StringUtils;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class FilterApplication implements Filter {
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        if (req.getMethod().equals("POST")) {
+            if (StringUtils.isNullOrEmpty(req.getParameter("name"))
+                    && StringUtils.isNullOrEmpty(req.getParameter("date"))
```

Comment:
It seems, we should have `||` instead of `&&` heresince we need to check, whether at least one of the parameters is empty. I think, it also helps to fix your error